### ISSUE TITLE
SteamCMD does not support spaces in +force_install_dir

### ISSUE THREAD

**Issue by Forlix** [ValveSoftware/Source-1-Games#350]
When providing an absolute path via commandline such as

steamcmd.exe +login anonymous +force_install_dir "C:\Program Files (x86)\Valve SrcDS\cstrike1" +app_update "232330" +exit

SteamCMD will create the folder "C:\program\" and install there instead.

**Comment by kisak-valve** [comment]
This issue is out-of-scope for this issue tracker. Closing.

**Comment by cryptearth** [comment]
well - would like to know how forlix got this issue in the firstplace
command.exe parses quotes before executing any command - so "C:\Program Files (x86)\Valve SrcDS\cstrike1" will get parsed into ONE string before steamcmd is even executed - the reported issue just can't happen with a normal working system - there obvious was a whole lot other crap wrong with this machine

**Comment by Forlix** [comment]
As @fletcherdvalve noted, SteamCMD simply fed those strings to its own custom command prompt with the quotes having been removed by command.exe as you stated.
For me the primary way of launching was through CreateProcess (Windows API) where I passed the command line string directly, facing the same issue. Back then I switched to passing the commands through standard input instead. Around March this year after a certain update that method stopped working however (SteamCMD now seems to bypass standard input), so I tried the command line method again and lo and behold it worked - it looks like they finally fixed this due to @egloeckle referencing it.
There are some hints to this as well when typing "help commandline".
